AnnoyancesFilter/Other/sections/self-promo.txt

```diff
@@ -5167,3 +5167,6 @@ blog.goo.ne.jp##.navbar__link--bottom
 eiga.com##.header-pure-banner
 !
 ||healthy-vape.shop/user_data/packages/*/img/banner/bnr_debut_gray.png
+
+! Banners within product listings.
+petbarn.com.au##[class^="ProductList"] > [class^="RecommendationBanner"]:upward(1)
```

Comment:
Hi.
In general the rule is ok, but when possible, better use native CSS instead of `:upward()`.
In this case can be used regular CSS selector:

```adblock
petbarn.com.au##div[class^="ProductGrid_productsGrid__"] > div[data-tile-type="ad"]
```

Added it + `petbarn.com.au##div[data-pa-banner-container]` (blinking empty leftover at the top of the page in categories) + blocked analytics.